pkg/util/podstore/podstore.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podstore
+
+import (
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+const (
+	DeleteQueueName = "pod_delete"
+	PodIPIndex      = "PodIP"
+	DelayTime       = time.Minute * 5
+)
+
+// for testing
+var delayTime = DelayTime
```

Comment:
Lately we are trying to use virtual / fake clocks for this purpose.
The way you do this is by using the `k8s.io/utils/clock` package, and by having a private constructor that takes a `clock.Clock` object as a parameter:

```go
func newPodStoreWithClock(podInformer cache.SharedIndexInformer, clock clock.Clock) *PodStore {
        // ...
}

func NewPodStore(podInformer cache.SharedIndexInformer) *PodStore {
        return newPodStoreWithClock(podInformer, clock.RealClock{})
}
```

in unit tests you use this clock: https://pkg.go.dev/k8s.io/utils/clock/testing#FakeClock. You can advance the clock as needed.

Reply:
Thanks for the suggestion. Updated.